tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+        if(($file.Name.ToLower() -eq ".git") -or -not (Split-Path $TargetPath)){
+            $foundRoot = $true
+        }
+    }
+
+    return @(
+        $foundConfig, $foundRoot
+    )
+}
+
+<#
+.SYNOPSIS
+Gets the relative path of the assets json within the target repo.
+
+.PARAMETER Config
+The config
+#>
+Function AscendToRepoRoot {
+    param(
+        [Parameter(Mandatory = $false)]
+        [string] $StartPath
+    )
+    $pathForManipulation = $StartPath
+    if(Test-Path $StartPath -PathType Leaf){
+        $pathForManipulation = Split-Path $pathForManipulation
+    }
+
+    $foundConfig, $reachedRoot = Evaluate-Target-Dir -TargetPath $pathForManipulation
+
+    while (-not $reachedRoot){
+        $pathForManipulation, $remainder = Split-Path $pathForManipulation
+
+        $foundConfig, $reachedRoot = Evaluate-Target-Dir -TargetPath $pathForManipulation
+    }
+
+    if ($reachedRoot){
+        return $pathForManipulation
+    }
+    else {
+        throw "Unable to the root of the git repo."
+    }
+}
+
+
+<#
+.SYNOPSIS
+Traverses up from a provided target directory to find a assets JSON, parses it, and returns the location and JSON content.
+
+.DESCRIPTION
+Traverses upwards until it hits either a `.git` folder or a `assets.json` file. Throws an exception if it can't find a assets.json before it hits root.
+
+.PARAMETER TargetPath
+Optional argument specifying the directory to start traversing up from. If not provided, current working directory will be used.
+#>
+Function Resolve-AssetsJson {
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $TargetPath
+    )
+    $pathForManipulation = $TargetPath
+    $foundConfig = $false
+    $reachedRoot = $false
+
+    if(-not $TargetPath){
+        $pathForManipulation = Get-Location
+    }
+    
+    $foundConfig, $reachedRoot = Evaluate-Target-Dir -TargetPath $pathForManipulation
+
+    while (-not $foundConfig -and -not $reachedRoot){
+        $pathForManipulation, $remainder = Split-Path $pathForManipulation
+
+        $foundConfig, $reachedRoot = Evaluate-Target-Dir -TargetPath $pathForManipulation
+    }
+
+    if ($foundConfig){
+        $discoveredPath = Join-Path $pathForManipulation "assets.json"
+    }
+    else {
+        throw "Unable to locate assets.json"
+    }
+
+    # path to assets Json
+    $config = (Get-Content -Path $discoveredPath | ConvertFrom-Json)
+    Add-Member -InputObject $config -MemberType "NoteProperty" -Name "AssetsJsonLocation" -Value "$discoveredPath"
+
+    $relPath = AscendToRepoRoot -StartPath $discoveredPath
+    if($relPath){
+        try {
+            Push-Location $relPath
+            $relPath = Resolve-Path -Relative -Path $discoveredPath
+
+            # relpaths are returned with ".\<blah>"
+            # given that, we need to get rid of it. This has possiiblity for bugs down the line.
+            $relPath = $relPath.Substring(2)
```

Comment:
Just in some weird case where it isn't prefixed with `./` (I don't know how different OSes or future powershell versions will behave, perhaps this is more defensive and also more clear:

```
            $relPath = -replace "^\.\/"
```